src/games/strategy/triplea/ui/MapRouteDrawer.java

```diff
@@ -149,10 +145,13 @@ private void drawCustomCursor(final Graphics2D graphics, final RouteDescription
    */
   private void drawDirectPath(final Graphics2D graphics, final Point start, final Point end, final int xOffset,
       final int yOffset, final double scale) {
-    drawLineWithTranslate(graphics, new Line2D.Float(start, end), xOffset,
-        yOffset, scale);
-    if (start.distance(end) > arrowLength) {
-      drawArrow(graphics, start, end, xOffset, yOffset, scale);
+    final Point[] points = routeOptimizer.getTranslatedRoute(start, end);
+    for (Point[] newPoints : routeOptimizer.getAllPoints(points)) {
+      drawLineWithTranslate(graphics, new Line2D.Float(newPoints[0], newPoints[1]), xOffset,
+          yOffset, scale);
+      if (newPoints[0].distance(newPoints[1]) > arrowLength) {
```

Comment:
some commentary to explain the logic and reasoning would be helpful. The heavy usage of array indexes is a bit troubling, but meh..